lib/site-builder.js

```diff
@@ -70,18 +71,31 @@ SiteBuilder.prototype.spawn = function(path, args) {
 };
 
 SiteBuilder.prototype.build = function(done) {
-  var that = this;
-  var doBuild = function(lockedOperationDone) {
-    fs.exists(that.opts.sitePath, function(exists) {
-      (exists ? that.syncRepo() : that.cloneRepo())
-        .then(function() { return that.checkForFile('_config.yml'); })
-        .then(function(useJekyll) {
-          return (useJekyll ? that._buildJekyll() : that._rsync());
-        })
-        .then(lockedOperationDone, lockedOperationDone);
+  var builder = this,
+      doBuild,
+      finish;
+
+  doBuild = function() {
+    return new Promise(function(resolve) {
+      fs.exists(builder.opts.sitePath, function(exists) {
+        resolve(exists ? builder.syncRepo() : builder.cloneRepo());
+      });
+    })
+    .then(function() { return builder.checkForFile('_config.yml'); })
+    .then(function(useJekyll) {
```

Comment:
`builder.updateLock.doLockedOperation(doBuild)` will propagate any rejections, and the `catch` block in the return statement at the end of `SiteBuilder.build` will handle the failure case before propagating the rejection further.